app/backends/workflow-orchestration.ts

```diff
@@ -81,23 +84,91 @@ async function _handleWorkItemResults(
 }
 
 /**
- * Creates a work item that uses all the output of the previous step.
+ * Read a STAC catalog and return the item links. This does not handle sub-catalogs. This function 
+ * makes assumptions based on the Harmony STAC directory layout for services inputs/outputs and
+ * is only intended to be used when aggregating service outputs into a single catalog.
+ * @param catlogPath - the path to the catalog
+ */
+async function getLinksFromCatalog(catalogPath: string): Promise<StacItemLink[]> {
+  const baseDir = path.dirname(catalogPath).replace(env.hostVolumePath, PATH_TO_CONTAINER_ARTIFACTS);
+  const text = (await fs.readFile(catalogPath)).toString();
+  const catalog = JSON.parse(text);
+  const links: StacItemLink[] = [];
+  for (const link of catalog.links) {
+    if (link.rel === 'item') {
+      // make relative path absolute
+      const { href } = link;
+      link.href = `${baseDir}/${path.normalize(href)}`;
+      links.push(link);
+    }
+  }
+
+  return links;
+}
+
+/**
+ * Creates a work item that uses all the output of the previous step. This function assumes that
+ * all work items for the previous step are completed. It also relies on the convention that
+ * services write out their results as STAC catalogs with the following path
+ * `/tmp/<JOB_ID>/<WORK_ITEM_ID>/outputs /catalogN.json`
+ * where N is from 0 to the number of results - 1.
  * 
  * @param tx - The database transaction
  * @param currentWorkItem - The current work item
  * @param nextStep - the next step in the workflow
- * @param results - an array of paths to STAC catalogs
+ * @param results - an array of paths to STAC catalogs from the last worked item
  */
 async function createAggregatingWorkItem(
-  tx: Transaction, currentWorkItem: WorkItem, nextStep: WorkflowStep, results: string[],
+  tx: Transaction, currentWorkItem: WorkItem, nextStep: WorkflowStep,
 ): Promise<void> {
-  // TODO use ALL outputs from the prior step here (HARMONY-961)
-  // just use the first result of the current output for now
+  const itemLinks: StacItemLink[] = [];
+  // get all the previous results
+  const workItemCount = await workItemCountForStep(tx, currentWorkItem.jobID, nextStep.stepIndex - 1);
+  let page = 1;
+  let processedItemCount = 0;
+  while (processedItemCount != workItemCount) {
```

Comment:
Possible infinite loop if this function is called before all of the results are available or if one of the database calls doesn't return all the items we expect, or some bug. We probably want some kind of safe guard to throw an error to prevent the infinite loop.